### PR TITLE
Added code action to insert missing top level signature

### LINKED ISSUE #725: Add a quick fix for top level bindings with no type signature
@Gurkenglas: 
> Top-level binding with no type signature:  isInSourceDir :: Foldable t1 => (t2 -> t3 -> Bool) -> t3 -> t1 t2 -> Bool" <- Could this provide a quickfix?

Would be a nice addition to `GhcMod.hs`

### PR DESCRIPTION
**Pull request by lorenzo**
fixes #725

### CODE REVIEW

File: src/Haskell/Ide/Engine/Plugin/GhcMod.hs [hunk 1]
```diff
@@ -1,4 +1,3 @@
-{-# LANGUAGE CPP                 #-}
```
Comment (bubba) [inline src/Haskell/Ide/Engine/Plugin/GhcMod.hs]: 💯

File: test/functional/FunctionalCodeActionsSpec.hs [hunk 2]
```diff
@@ -278,6 +278,29 @@ spec = describe "code actions" $ do
             \  where\n\
             \    stuff (A a) = A (a + 1)\n"
 
+  describe "missing top level signature code actions" $
+    it "Adds top level signature" $
+      runSession hieCommand fullCaps "test/testdata/" $ do
+        doc <- openDoc "TopLevelSignature.hs" "haskell"
+
+        _ <- waitForDiagnosticsSource "ghcmod"
+        cas <- map (\(CACodeAction x)-> x) <$> getAllCodeActions doc
```
Comment (bubba) [inline test/functional/FunctionalCodeActionsSpec.hs]: Can you reuse `fromAction` here?
https://github.com/haskell/haskell-ide-engine/blob/deae46c393988872c212e0ab595dbeb28e78bd60/test/functional/FunctionalCodeActionsSpec.hs#L304-L306

File: test/functional/FunctionalCodeActionsSpec.hs [hunk 3]
```diff
@@ -278,6 +278,29 @@ spec = describe "code actions" $ do
             \  where\n\
             \    stuff (A a) = A (a + 1)\n"
 
+  describe "missing top level signature code actions" $
+    it "Adds top level signature" $
+      runSession hieCommand fullCaps "test/testdata/" $ do
+        doc <- openDoc "TopLevelSignature.hs" "haskell"
```
Comment (bubba) [inline test/functional/FunctionalCodeActionsSpec.hs]: Did you forget to commit this file?

Comment (lorenzo) [reply test/functional/FunctionalCodeActionsSpec.hs]: I did, fixed!

File: test/functional/FunctionalCodeActionsSpec.hs (hunk 2 above)
Comment (lorenzo) [reply test/functional/FunctionalCodeActionsSpec.hs]: cool, I'll try that

**Comment by alanz** [comment]
LGTM

**Pull request merged**